userspace/engine/ruleset.h

```diff
@@ -72,26 +70,7 @@ class falco_ruleset
 
 private:
 
-	class filter_wrapper {
-	public:
-		std::string source;
-		std::string name;
-		std::set<std::string> tags;
-		std::shared_ptr<gen_event_filter> filter;
-		std::set<uint16_t> evttypes()
-		{
-			// todo(jasondellaluce,leogr): temp workaround, remove when fixed in libs
-			if(source == "syscall" || source == "k8s_audit")
-			{
-				return filter->evttypes();
```

Comment:
So this is the key change that reduces memory usage right? Not calling gen_event_filter::evttypes() at all, which prevents the set from being populated.

I assume a follow-on PR will remove all the evttypes() code from libs, right?

Reply:
See the comment above about the cleanup PR in libs. However, the key solution here is to **not** store the event types set in each filtercheck. The sets used for the calculations are deallocated at each function call.